File: 백준/Silver/10828. 스택/스택.py
import sys

n = int(sys.stdin.readline())
s = []

for _ in range(n):
    i = sys.stdin.readline().split()

    if i[0] == 'push':
        s.append(int(i[1]))

    elif i[0] == 'pop':
        if len(s) == 0:
            print(-1)
        else:
            print(s.pop())

    elif i[0] == 'size':
        print(len(s))

    elif i[0] == 'empty':
        if len(s) == 0:
            print(1)
        else:
            print(0)

    elif i[0] == 'top':
        if len(s) == 0:
            print(-1)
        else:
            print(s[-1])

File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#include <string.h>

int stack[10000];
int top = -1;

int is_empty() { return top == -1; }

void push(int data) { stack[++top] = data; }

int pop() { return stack[top--]; }

int main(void) {
    int n, item;
    char cmd[100] = "";
    scanf("%d", &n);
    while (n--) {
        scanf("%s", cmd);
        if (strcmp(cmd, "push") == 0) {
            scanf("%d", &item);
            push(item);
        } else if (strcmp(cmd, "pop") == 0) {
            if (is_empty()) {
                printf("-1\n");
            } else {
                printf("%d\n", pop());
            }
        } else if (strcmp(cmd, "size") == 0) {
            printf("%d\n", top + 1);
        } else if (strcmp(cmd, "empty") == 0) {
            if (is_empty()) {
                printf("1\n");
            } else {
                printf("0\n");
            }
        } else if (strcmp(cmd, "top") == 0) {
            if (is_empty()) {
                printf("-1\n");
            } else {
                printf("%d\n", stack[top]);
            }
        }
    }
}